### PR TITLE
Add `functools.wraps` support for `is_overridden`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/8273

Some very minor doc fixes are included

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by carmocca** [comment]
@awaelchli Do we need this in the bugfix branch? `is_overridden` was refactored and the changes aren't there.

**Review by awaelchli — APPROVED** [review]
oh this is very nice! I would have never figured this out <3

**Review by tchaton — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Comment by awaelchli** [comment]
> @awaelchli Do we need this in the bugfix branch? is_overridden was refactored and the changes aren't there.

Don't think so, because calling the methods when they are not implemented will do nothing. So it's not noticable.

File: tests/utilities/test_model_helpers.py [hunk 1]
```diff
@@ -37,13 +37,40 @@ class TestModel(BoringModel):
...
+            def wrapper():
+                fn()
+
+            return wrapper
+
+        def bar(self):
+            return 2
+
+    # `functools.wraps()` support
+    assert not is_overridden("foo", WrappedModel(), parent=TestModel)
```
Comment (Borda) [inline tests/utilities/test_model_helpers.py:71]: I feel like this shall mark as overwritten as a wrapped can significantly change its functionality...

Comment (carmocca) [reply tests/utilities/test_model_helpers.py:71]: The current implementation assumes that the wrapper will be the same both for the instance and the parent

Not sure whether we could check if the wrappers are different

Comment (Borda) [reply tests/utilities/test_model_helpers.py:71]: can you detect what wrapper was used and then compare the wrapper itself?

Comment (carmocca) [reply tests/utilities/test_model_helpers.py:71]: Can detect that a wrapper was used (current PR changes) but no idea about comparing the wrappers

Comment (tchaton) [reply tests/utilities/test_model_helpers.py:71]: Could we check if the wrapper is from `Lightning` or `users` ?

Comment (tchaton) [reply tests/utilities/test_model_helpers.py:71]: Might be worth for another PR. I believe this is a pretty rare use-case.

**Pull request merged**